pyglet/graphics/vertexdomain.py

```diff
@@ -292,33 +336,60 @@ def migrate(self, domain):
         self.domain.allocator.dealloc(self.start, self.count)
         self.domain = domain
         self.start = new_start
+        self._cache_version = None
 
-        for version in self._cache_versions:
-            self._cache_versions[version] = None
+    def get_setter_params(self, name):
+        """
+        Returns a tuple of params which can be passed to
+        `vertex_list_set_attribute_values` along with a data tuple to set values
+        for the attribute.
+
+        These params become invalid when the VertexList migrates to a different
```

Comment:
This might be a blocker, unfortunately. For the default Batch, users should be able to add/remove/migrate things between Batches transparently.

Reply:
This is safe for `Sprite` to use internally, right?

`Sprite._batch` and `Sprite._vertex_list` are private.  For a user to migrate a Sprite between batches, they have to assign to `Sprite.batch`, `Sprite.program`, etc.  And all those properties call `VertexList.get_setter_params`.

If a user is using `VertexDomain` / `VertexList` directly, they can safely avoid `get_setter_params` and instead use the same API that sprites do today: get a `BufferObjectRegion` via `VertexList.__getattr__` or the generated properties on `_vertexlist_class`.